app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/monsters/detail/MonsterRewardFragment.java

```diff
@@ -65,13 +67,21 @@ public View onCreateView(LayoutInflater inflater, ViewGroup parent,
         return recyclerView;
     }
 
+    public static MonsterRewardFragment newInstance(Rank rank) {
+        MonsterRewardFragment fragment = new MonsterRewardFragment();
+        fragment.setArguments(new BundleBuilder()
+                .putSerializable(ARG_RANK, rank)
+                .build());
+        return fragment;
+    }
+
     /**
      * Set the rewards to be displayed in the fragment
      * @param rewards items be of type Reward.
      */
     public void setItems(List<MonsterRewardView> rewards) {
-        Bundle bundle = this.getArguments();
-        List<Object> rewardsWithHeaders = populateHeaders(rewards, (Rank) bundle.getSerializable("rank"));
+
+        List<Object> rewardsWithHeaders = populateHeaders(rewards, (Rank) getArguments().getSerializable(ARG_RANK));
```

Comment:
I'm ok with committing as it is, but for these kinds of things its more readable if you split it into separate lines. You'd populate a Rank variable before using it.